hnn_core/gui/_viz_manager.py

```diff
@@ -229,7 +256,16 @@ def _plot_on_axes(b, widgets_simulation, widgets_plot_type,
 
 def _clear_axis(b, widgets, data, fig_idx, fig, ax, widgets_plot_type,
                 existing_plots, add_plot_button):
+    # clear all potential linked axes before any modifications.
+    _disable_align_fig_xaxis(fig)
     ax.clear()
+    delattr(ax, 'align_x')
+
+    # remove attached colorbar if exists
+    if hasattr(fig, f'_cbar-ax-{id(ax)}'):
+        getattr(fig, f'_cbar-ax-{id(ax)}').ax.remove()
+        delattr(fig, f'_cbar-ax-{id(ax)}')
```

Comment:
Kind of curious what happens if you don't delete it, does it just maintain a reference to the missing colorbar?

Reply:
I guess so. Deleting is more for clearness here and the chance that some other variable takes that memory block should be neglectable.